### PR TITLE
Drop support for python 2.7

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Removes the support for Python 2.7 in the Datadog Lambda Layer for Python

### Motivation

It is no longer maintained and AWS is dropping the support as well.

### Testing Guidelines

Updated the snapshots and integration tests.

### Additional Notes

N/A

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

**Comment by duncanista** [comment]
This should be merge by September 30th

**Comment by maxday** [comment]
This looks good! There are also some places where we're doing try/catch around import. Since we don't support 2.7 anymore, maybe those can be simplified. Example : https://github.com/DataDog/datadog-lambda-python/blob/main/tests/test_extension.py#L7

**Comment by duncanista** [comment]
Unit and integration tests for 2.7 will be removed from the settings.

**Review by maxday — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -19,23 +19,20 @@
...
     ],
     keywords="datadog aws lambda layer",
     packages=["datadog_lambda"],
-    python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*, <4",
+    python_requires=">=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*, <4",
     install_requires=[
         "datadog==0.41.0",
         "ddtrace==0.53.0",
         "wrapt==1.11.2",
         # If building for Python 3, use the latest version of setuptools
```
Comment (agocs) [inline setup.py:34]: You can probably remove this comment.

File: datadog_lambda/patch.py [hunk 2]
```diff
@@ -17,15 +17,11 @@
...
 
 logger = logging.getLogger(__name__)
 
-if sys.version_info >= (3, 0, 0):
-    httplib_module = "http.client"
-    from collections.abc import MutableMapping
-else:
-    httplib_module = "httplib"
-    from collections import MutableMapping
+httplib_module = "http.client"
```
Comment (agocs) [inline datadog_lambda/patch.py:24]: You could probably make this local to `_patch_httplib`. You might also want to consider changing that function's name and removing Python 2 from its docstring.

**Comment by duncanista** [comment]
Unit and integration tests for 2.7 were removed from Settings > Branches > Main.

**Comment by duncanista** [comment]
Renaming the branch.

**Pull request merged**